data/dataset.py

```diff
@@ -0,0 +1,115 @@
+import random
+
+import torch
+import numpy as np
+from torch.utils.data import Dataset
+from datasets import Dataset as HFDataset
+
+from data.quantizer import MidiQuantizer
+from data.tokenizer import MidiEncoder
+from data.augmentation import change_speed, pitch_shift
+from data.masking import AwesomeMasks
+
+
+class MidiDataset(Dataset):
+    def __init__(
+            self, 
+            dataset: HFDataset, 
+            tokenizer: MidiEncoder, 
+            augmentation_probability: float = 0.0, 
+            masking_probability: float = 0.15
+        ):
+        super().__init__()
+
+        self.dataset = dataset
+        self.quantizer = MidiQuantizer(7, 7, 7)
+        self.tokenizer = tokenizer
+        self.masks = AwesomeMasks()
+
+        self.augmentation_probability = augmentation_probability
+        self.masking_probability = masking_probability
+
+    def __len__(self):
+        return len(self.dataset)
+    
+    def apply_augmentation(self, record: dict):
+        # shift pitch augmentation
+        if random.random() < self.augmentation_probability:
+            # max shift is octave down or up
+            shift = random.randint(1, 12)
+            record["pitch"] = pitch_shift(record["pitch"], shift)
+
+        # change tempo augmentation
+        if random.random() < self.augmentation_probability:
+            record["dstart"], record["duration"] = change_speed(record["dstart"], record["duration"])
+            # change bins for new dstart and duration values
+            record["dstart_bin"] = np.digitize(record["dstart"], self.quantizer.dstart_bin_edges) - 1
+            record["duration_bin"] = np.digitize(record["duration"], self.quantizer.duration_bin_edges) - 1
+
+        return record
+    
+    def apply_masking(self, token_ids: np.ndarray, record: dict):
+        # masking, adds new key to record dict called masked
+        masked, _ = self.masks.apply(record, p=self.masking_probability)
+
+        blank_idx = self.tokenizer.token_to_id["<blank>"]
```

Comment:
```suggestion
        blank_idx = self.tokenizer.token_to_id["<mask>"]
```

"blank" sounds like a white space to me, let's be more explicit